File: q0538/q0538.py
class Solution:
    def convertBST(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        self.currSum = 0
        self.dfsHelper(root)
        return root
    
    def dfsHelper(self, node):
        if not node: return
        self.dfsHelper(node.right)
        self.currSum += node.val
        node.val = self.currSum
        self.dfsHelper(node.left)
        return 

File: q0538/q0538.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    private int currSum = 0;

    public TreeNode convertBST(TreeNode root) {
        dfsHelper(root);
        return root;
    }

    private void dfsHelper(TreeNode node) {
        if (node == null)
            return;
        dfsHelper(node.right);
        currSum += node.val;
        node.val = currSum;
        dfsHelper(node.left);
        return;
    }
}